src/data/recorder.ts

```diff
@@ -324,3 +328,17 @@ export const getDisplayUnit = (
 
 export const isExternalStatistic = (statisticsId: string): boolean =>
   statisticsId.includes(":");
+
+let recordedEntitiesCache: RecordedEntities | undefined;
+
+export const getRecordedEntities = async (
+  hass: HomeAssistant
+): Promise<RecordedEntities> => {
+  if (recordedEntitiesCache) {
+    return recordedEntitiesCache;
```

Comment:
This means new entities will never show up, I don't think we should do this, it will cause users to wonder why their new entity is not showing up in the picker...

Reply:
Oof, right. Maybe invert the list when we fetch it, and cache the list of things which are currently excluded?

I didn't start with caching, but I thought it would be nice as this could be kind of a big list to retrieve repeatedly, and it doesn't change much in a meaningful way (at least without someone editing their config.yaml and restarting). 

Maybe also inquire about inverting the polarity of the API, to fetch excluded entities instead of recorded ones. I feel like that's generally going to be a much smaller list in the average usecase.

Reply:
I think it might be easier to just get the filters that are set by the user in config, we can then use https://github.com/home-assistant/frontend/blob/8cad37af8f97dd3796966e91c63e96791591ee34/src/common/entity/entity_filter.ts#L19 to filter the entities...

Reply:
Maybe leave that comment in the core thread so we can collect all the ideas in one place?

I do kind of like that we're now getting visibility into exactly what the backend thinks is recorded, instead of reproducing the filtering logic ourselves, as we've already identified a bug as a result of that. But this could work too.

Reply:
Yeah, we don't support things like globs in frontend...

Reply:
Ok flipped the logic around a bit. I still cache the result, but at the moment I get back the api response, I also cache a list of any entity that is present in hass.states, but not in the recorded response. And then use that to exclude from the picker. 

So any new entities created after the cache was fetched would not be excluded. 

bdraco also suggested we can use this information to populate the picker with entities which are orphaned/missing from the states table, but still have history in the database. Haven't done that yet, but the plumbing is there to do so.

Reply:
> I think hass.states is the right thing to check against right? I don't need to do anything additionally with entity registry?

Yeah that should be fine 👍